crates/autopilot/src/run_loop.rs

```diff
@@ -443,6 +452,34 @@ impl RunLoop {
     async fn save_competition(&self, competition: &Competition) -> Result<()> {
         self.database.save_competition(competition).await
     }
+
+    /// Removes orders that are currently being settled to avoid solvers trying
+    /// to fill an order a second time.
+    async fn remove_in_flight_orders(&self, mut auction: Auction) -> Auction {
+        let prev_settlement = self.in_flight_orders.lock().unwrap().tx_hash;
+        let tx_receipt = self.web3.eth().transaction_receipt(prev_settlement).await;
```

Comment:
What happens if in the first iteration the `tx_hash` is initialized with `Default::default()`?

Reply:
Then `transaction_receipt()` would try to find the block that settled tx hash `0x0000` which does not exist so we fallback to assuming the previous auction was settled in block `u64::MAX` so we discard all in-flight orders we have at the time (which are 0 due to `Default::default()`).